File: Server/src/main/java/com/api/server/foodb/CompoundsHealthEffect.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "compounds_health_effects")
public class CompoundsHealthEffect {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "compound_id", nullable = false)
    private Integer compoundId;

    @NotNull
    @Column(name = "health_effect_id", nullable = false)
    private Integer healthEffectId;

    @Size(max = 255)
    @Column(name = "orig_health_effect_name")
    private String origHealthEffectName;

    @Size(max = 255)
    @Column(name = "orig_compound_name")
    private String origCompoundName;

    @Lob
    @Column(name = "orig_citation")
    private String origCitation;

    @NotNull
    @Lob
    @Column(name = "citation", nullable = false)
    private String citation;

    @Size(max = 255)
    @NotNull
    @Column(name = "citation_type", nullable = false)
    private String citationType;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "creator_id")
    private Integer creatorId;

    @Column(name = "updater_id")
    private Integer updaterId;

    @Column(name = "source_id")
    private Integer sourceId;

    @Size(max = 255)
    @Column(name = "source_type")
    private String sourceType;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/CompoundsHealthEffect.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("compounds_health_effects")]
[Index("SourceId", "SourceType", Name = "index_compounds_health_effects_on_source_id_and_source_type")]
public class CompoundsHealthEffect
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("compound_id")] public int CompoundId { get; set; }

    [Column("health_effect_id")] public int HealthEffectId { get; set; }

    [Column("orig_health_effect_name")]
    [StringLength(255)]
    public string? OrigHealthEffectName { get; set; }

    [Column("orig_compound_name")]
    [StringLength(255)]
    public string? OrigCompoundName { get; set; }

    [Column("orig_citation", TypeName = "mediumtext")]
    public string? OrigCitation { get; set; }

    [Column("citation", TypeName = "mediumtext")]
    public string Citation { get; set; } = null!;

    [Column("citation_type")]
    [StringLength(255)]
    public string CitationType { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }

    [Column("creator_id")] public int? CreatorId { get; set; }

    [Column("updater_id")] public int? UpdaterId { get; set; }

    [Column("source_id")] public int? SourceId { get; set; }

    [Column("source_type")] public string? SourceType { get; set; }
}